### PR TITLE
Dispatch ALTER SYSTEM command

### PR DESCRIPTION
**Pull request by ashwinstar**
`ALTER SYSTEM` only modified the GUC on coordinator. Changing it to dispatch the command to segments as well. Would be ideal in long run for `ALTER SYSTEM` similar to `gpconfig` to have `coordinator_only` option. But for now it seems more natural to set GUC on all nodes instead of only coordinator. Given this command directly works directly on filesystem and not transactional, if fails on a segment, the modification will still remain on other segments. I believe this is no different from `gpconfig` failing half way.

Also, in this PR then changing disable/enable autovacuum test to take effect on segments along with coordinator. That hopefully should help to fix the flakiness we are seeing in CI post the autovacuum feature was enabled for catalog on segments.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by ashwinstar** [comment]
One thought I am having is for `ALTER SYSTEM` we should be able to detect coordinator_only gucs maybe using the `GUC_GPDB_NO_SYNC` flag and avoid dispatches for them. That way in tests and user for example FTS gucs will not be dispatched and test don't have to explicitly specify the flag.

File: src/test/isolation2/expected/terminate_in_gang_creation.out [hunk 1]
```diff
@@ -165,9 +165,9 @@ SELECT gp_inject_fault_infinite('before_orphaned_check', 'reset', dbid) FROM gp_
 --------------------------
  Success:                 
 (1 row)
-ALTER SYSTEM RESET gp_dtx_recovery_interval;
+2: ALTER SYSTEM RESET gp_dtx_recovery_interval;
```
Comment (asimrp) [inline src/test/isolation2/expected/terminate_in_gang_creation.out:168]: Why is this change needed?

File: src/backend/tcop/utility.c [hunk 2]
```diff
@@ -782,6 +782,11 @@ standard_ProcessUtility(PlannedStmt *pstmt,
 		case T_AlterSystemStmt:
 			PreventInTransactionBlock(isTopLevel, "ALTER SYSTEM");
 			AlterSystemSetConfigFile((AlterSystemStmt *) parsetree);
+			if (Gp_role == GP_ROLE_DISPATCH)
+				CdbDispatchUtilityStatement((Node *) parsetree,
```
Comment (asimrp) [inline src/backend/tcop/utility.c:786]: If the dispatch call is moved inside `AlterSystemSetConfigFile`, it is possible to check if `GUC_GPDB_NO_SYNC` is set for the desired config option.

**Review by asimrp — APPROVED** [review]
Thank you for doing this, I love using ALTER SYSTEM command in the tests and now will continue to do so with even more excitement :)

File: src/test/isolation2/expected/terminate_in_gang_creation.out (hunk 1 above)
Comment (ashwinstar) [reply src/test/isolation2/expected/terminate_in_gang_creation.out:168]: The existing gangs is cleaned due to primary restart by the test. Previously, the query was run only on coordinator and hence it was fine. But if we try to dispatch to segments it fails on first query. Hence, need to create fresh connection for this now.

File: src/backend/tcop/utility.c (hunk 2 above)
Comment (ashwinstar) [reply src/backend/tcop/utility.c:786]: I gave more thought about this aspect and realized `GUC_GPDB_NO_SYNC` doesn't really convey COORDINATOR_ONLY GUCs. For example `autovacuum`, `log_statement` and many more gucs are listed in `unsync_guc_name.h` but when modifying the value of them I wish to modify on coordinator as well as segments. These might be gucs which can't be set on session level and hence `GUC_GPDB_NO_SYNC` but via conf file the change is allowed. So, we really need to come up with new flag to mark `COORDINATOR_ONLY` and then can avoid the dispatch using the same. Plus, moving the dispatch inside `AlterSystemSetConfigFile()` feels weird as well. Hence, leaving it the way we have it can think if enhancing the same in future.

**Comment by lisakowen** [comment]
@ashwinstar - does this PR add ALTER SYSTEM support for users of greenplum 7?

**Comment by ashwinstar** [comment]
> @ashwinstar - does this PR add ALTER SYSTEM support for users of greenplum 7?

Logically, support existed even before. This PR modifies the `ALTER SYSTEM` behavior and makes it more MPP aware. I am not sure what we mention in documentation for ALTER SYSTEM currently. With this PR now any GUC modification using the `ALTER SYSTEM` command happens both on coordinator and segments. Previously, it was only on coordinator. So, still the functionality is minimal and not complete in semantics with `gpconfig`. Though its better than before. Thank you watching and question.

**Comment by lisakowen** [comment]
the 6.x docs state that we don't support ALTER SYSTEM.  should we plan to document this (with caveats) for greenplum 7, or would it be best to keep as unsupported?

**Comment by ashwinstar** [comment]
> the 6.x docs state that we don't support ALTER SYSTEM. should we plan to document this (with caveats) for greenplum 7, or would it be best to keep as unsupported?

I think we can keep it as unsupported for now for GPDB7 as well. We have not heard much demand from field or community on need for this hence being conservative for it and only keeping it for testing for now is okay. Once we fully add all MPP features for it or see demand for it we can change the documentation.
